src/backends/onnx/onnx_backend.cc

```diff
@@ -545,49 +540,125 @@ OnnxBackend::Context::ValidateOutputs(
   return Status::Success;
 }
 
-Status
+namespace {
+
+#ifdef TRTIS_ENABLE_STATS
+#define FAIL_ALL_AND_RETURN_IF_ERROR(REQUESTS, RESPONSES, S)        \
```

Comment:
I think these macros would be good to but into backend.h... and I think TF backend is doing the wrong thing WRT updating the stats when it does this equivalent thing. Can you update TF backend to use this macro as well.